devicetree: instance macros for `STRING_TOKEN`

JordanYates opened the pull request:
Adds instance based macros for the `DT_STRING_TOKEN` and `DT_STRING_TOKEN_UPPER` macros.
This can make `DT_INST_FOREACH_STATUS_OKAY` macros a little cleaner.

Commit:
devicetree: additional `STRING_TOKEN` macros

Add a variety of missing `DT_INST` and `_OR` variants for working with
`STRING_TOKEN` and `STRING_UPPER_TOKEN` macros.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

Commit:
spi: spi_b91: use new `DT_INST_STRING_TOKEN` macro

Use the dedicated `DT_INST_STRING_TOKEN` macro instead of manually
retrieving `DT_DRV_INST`.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

galak reviewed (dismissed):
Please add a test for these tests/lib/devicetree/api

Commit:
tests: devicetree: api: new `STRING_TOKEN` macros

Add tests for the new `STRING_TOKEN` macros.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

JordanYates commented:
> Please add a test for these tests/lib/devicetree/api

Additional missing macro variants + tests have been added

mbolivar-nordic commented on include/devicetree.h:
```diff
@@ -2471,6 +2471,26 @@
+ * @brief Like DT_INST_STRING_TOKEN(), but uppercased.
+ * @param inst instance number
+ * @param prop lowercase-and-underscores property string name
+ * @return the value of @p prop as a token, i.e. without any quotes
+ *         and with special characters converted to underscores
```
Can you please apply the same fix I sent here? https://github.com/zephyrproject-rtos/zephyr/pull/44237

mbolivar-nordic commented on include/devicetree.h:
```diff
@@ -868,6 +868,23 @@
+ * @param node_id node identifier
+ * @param prop lowercase-and-underscores property name
+ * @param default_value a fallback value to expand to
+ * @return the property's value or default_value
```
value as an uppercased token or

mbolivar-nordic commented on include/devicetree.h:
```diff
@@ -2761,6 +2761,17 @@
 	DT_STRING_TOKEN_OR(DT_DRV_INST(inst), name, default_value)
 
+/**
+ * @brief Tests if DT_DRV_COMPAT's has string property, returns uppercased
+ *        string token if found, otherwise returns default_value.
```
Like DT_INST_STRING_UPPER_TOKEN(), but with a fallback to default_value

mbolivar-nordic commented on tests/lib/devicetree/api/src/main.c:
```diff
@@ -2229,6 +2236,17 @@ static void test_string_token(void)
 		zassert_true(array_inst[i] != token_no_inst, "");
 	}
 
+	/* Test DT_INST_STRING_UPPER_TOKEN_OR when property is found */
```
It looks like this is missing a test for DT_INST_STRING_TOKEN and DT_INST_STRING_UPPER_TOKEN. We should have tests for every macro in devicetree.h, even if right now they're implemented in terms of something else. This is so that we avoid regressions in case that implementation detail changes in the future.

mbolivar-nordic reviewed (changes requested):
A few requests for docstring changes and some missing test cases. Otherwise LGTM.

Commit:
ztest: Add zassert_between_inclusive macro

Added macro to test that a value falls within a defined range, including
the provided end points.

Signed-off-by: Jordan Yates <jordan.yates@data61.csiro.au>

mbolivar-nordic commented:
@galak ping
